### PR TITLE
Feature: List Project Groups functionality

### PR DESCRIPTION
**Pull request by CGoodwin90**
# Checklist
- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated.
- [ ] Changelog entry has been written

Adds the functionality to the CLI to list groups associated to a specified project.

Currently utilizing & pending merge of machinery PR https://github.com/uselagoon/machinery/pull/23

### CODE REVIEW

**Review by shreddedbacon — CHANGES_REQUESTED** [review]
This will also need `make docs` run to make sure the documentation is updated

**Review by shreddedbacon — APPROVED** [review]
Very handy addition

File: cmd/list.go [hunk 1]
```diff
@@ -436,11 +437,66 @@ var listNotificationCmd = &cobra.Command{
...
+		lc := lclient.New(
+			lagoonCLIConfig.Lagoons[current].GraphQL,
+			lagoonCLIVersion,
+			&token,
+			debug)
+		projectGroups, err := l.GetProjectGroups(context.TODO(), cmdProjectName, lc)
+		handleError(err)
+
+		if len(projectGroups.Groups) == 0 {
+			output.RenderInfo(fmt.Sprintf("There are no projects in group '%s'", groupName), outputOptions)
```
Comment (sjerdo) [inline cmd/list.go:469]: Where is `groupName` defined? This message is confusing. I expect this message to be like "There are no groups for project '%s'" with the cmdProjectName as substitute

Comment (shreddedbacon) [reply cmd/list.go:469]: :see_no_evil: 
Will fix it up

Comment (shreddedbacon) [reply cmd/list.go:469]: Based on the condition though, this shouldn't ever be reached as projects should always have 1 group. Did you encounter this error?

Comment (sjerdo) [reply cmd/list.go:469]: Yeah I get this error message for all of our projects. We haven't assigned our projects to a group yet.

```
~ % lagoon list project-groups -p <PROJECT NAME>
Info: There are no projects in group ''
```

**Pull request merged**